### PR TITLE
[Git] Set env variable to en_US-UTF8

### PR DESCRIPTION
**Pull request by bogren**
Proposed change form #476. Need to test the change though, I'll give it a try tomorrow on a Travis machine.

### CODE REVIEW

**Comment by orta** [comment]
@bogren this looks great - wanna add a CHANGELOG to credit yourself then we're good to go 👍

**Comment by k0nserv** [comment]
@orta Is there a way to test backtick command line calls? Noticed that this doesn't have any tests

**Comment by orta** [comment]
This is used in a lot of places, check out the windows CIs, but alas this change does not work on windows, which we need to keep supporting, might need to think of another way of doing it

**Comment by bogren** [comment]
I'm looking for new possible solutions 🔎

**Comment by orta** [comment]
If it comes to it, we can also use `system` which allows setting env vars, and return the output of that

File: lib/danger/scm_source/git_repo.rb [hunk 1]
```diff
@@ -15,7 +15,9 @@ def diff_for_folder(folder, from: "master", to: "HEAD")
     end
 
     def exec(string)
-      `LANG=en_US.UTF-8 git #{string}`.strip
+      # See #476
+      ENV["LANG"] = "en_US.UTF-8"
+      `git #{string}`.strip
```
Comment (JuanitoFatas) [inline lib/danger/scm_source/git_repo.rb]: How about?

```ruby
Open3.popen3({"LANG" => "en_US.UTF-8"}, "git #{string}") { |_stdin, stdout, _stderr, _wait_thr| stdout.read.strip }
```

[Open3](http://ruby-doc.org/stdlib-2.0.0/libdoc/open3/rdoc/Open3.html#method-c-popen3)

Comment (orta) [inline lib/danger/scm_source/git_repo.rb]: is that synchronous?

Comment (orta) [inline lib/danger/scm_source/git_repo.rb]: ah yeah, looks like it, aye, I'd be happy with that instead

**Comment by JuanitoFatas** [comment]
Sorry, do we care about the error? Otherwise we can use popen2.

> ::popen2 is similer to ::popen3 except it doesn't make a pipe for the standard error stream.
> [Open3#popen2](http://ruby-doc.org/stdlib-2.0.0/libdoc/open3/rdoc/Open3.html#method-c-popen2)

**Comment by orta** [comment]
I think we were already throwing it away, so no

**Comment by JuanitoFatas** [comment]
Updated and I added a test to make sure 💪

**Comment by orta** [comment]
Great -yeah - this looks 💯  - thanks @bogren and @JuanitoFatas 👍

**Pull request merged**